### DISCUSSION TITLE
[Q] RabbitMQ durable queue

### DISCUSSION THREAD

**Discussion by Colomix** [spiral/roadrunner-plugins#167] [Q&A] [answered]
I want add job consume to queue with durable: true, but my service fail, because i can't set it from config

**Comment by rustatian** [comment]
Hey @Colomix , please, turn on debug logs and share the output with error. Also, could you please add your `.rr.yaml` configuration here. Thanks.

**Comment by Colomix** [comment]
https://github.com/spiral/roadrunner-plugins/blob/master/amqp/amqpjobs/rabbit_init.go
`q, err := channel.QueueDeclare(
		c.queue,
		false,
		false,
		c.exclusive,
		false,
		[nil,](url)
	)`

Second argument is durable, it is always false. I want consume to queue with setting durable is true. In RR 1.* similar code default set true, that's why my applications cannot communicate through the queue and I am deadlocked

`
amqp:
  addr: amqp://guest:guest@rabbitmq:5672/

jobs:
  pool:
    num_workers: 1
    max_jobs: 0
    allocate_timeout: 600s
    destroy_timeout: 600s

  pipelines:
    job-event-receiver:
      driver: amqp
      prefetch: 10
      queue: job.fsm.event.receiver
      exchange: default
      exchange-type: direct
      routing-key: job.fsm.event.receiver
      multiple_ack: false
      requeue_on_fail: false
  consume: [job-event-receiver]
`

ERROR:
handle_serve_command: Serve error:
        endure_start:
        endure_serve_internal: Function call error:
        endure_call_serve_fn: got initial serve error from the Vertex jobs.Plugin, stopping execution, error: jobs_plugin_serve:
        new_amqp_consumer:
        jobs_plugin_rmq_init: Exception (406) Reason: "PRECONDITION_FAILED - inequivalent arg 'durable' for queue 'job.fsm.event.receiver' in vhost '/': received 'false' but current is 'true'"

**Reply by rustatian** [reply]
That should not be the default behavior, but I'll move `durable` key to the pipeline options.

**Reply by rustatian** [reply]
@Colomix Thanks for the report, I've moved this discussion to the ticket.

**Comment by rustatian** [comment, marked as answer]
Fixed via #196
